contracts/src/tokens/memecoin.cairo

```diff
@@ -137,6 +159,72 @@ mod UnruggableMemecoin {
                 * MAX_SUPPLY_PERCENTAGE_TEAM_ALLOCATION.into()
                 / 100
         }
+
+        /// Sets the Merkle root for the contract.
+        /// This function updates the Merkle root stored in the contract's state.
+        /// It is essential for maintaining the integrity of the Merkle tree used in various contract functionalities.
+        
+        /// # Arguments
+        /// * `merkle_root` - The new Merkle root to be set, represented as a `felt252`.
+
+        fn set_merkle_root(ref self: ContractState, merkle_root: felt252) {
+            self.ownable.assert_only_owner();
+            self.initializable.initialize();
+            self.merkle_root.write(merkle_root);
+        }
+
+        /// Retrieves the current Merkle root from the contract.
+        /// This function allows the contract owner to obtain the current Merkle root stored in the contract's state.
+        /// The Merkle root is crucial for verifying proofs in various contract operations.
+
+        /// # Returns
+        /// * `felt252` - The current Merkle root stored in the contract.
+
+        fn get_merkle_root(self: @ContractState) -> felt252 {
+            //Getting the merkle root
+            self.ownable.assert_only_owner();
+            self.merkle_root.read()
+        }
+
+        /// Claims an airdrop for a specific account.
+        /// This function is part of the contract's state and is used to claim airdrops for accounts.
+        /// It involves a Merkle tree verification process to ensure the legitimacy of the claim.
+
+        /// # Arguments
+        /// * `to` - The address of the contract for which the airdrop is being claimed.
+        /// * `amount` - The amount of tokens to be airdropped, represented as a `u256`.
+        /// * `leaf` - A mutable leaf node in the Merkle tree, represented as a `felt252`.
+        /// * `proof` - A mutable span of `felt252` elements representing the Merkle proof.
+        fn claim_airdrop(
+            ref self: ContractState,
+            to: ContractAddress,
+            amount: u256,
+            mut leaf: felt252,
+            mut proof: Span<felt252>,
+        ) {
+            //Initializing the Merkletree
+            let mut merkle_tree: MerkleTree<Hasher> = MerkleTreeTrait::new();
+            //Pedersen Hashing of the ContractAddress and Amount
+            let to_felt252: felt252 = starknet::contract_address_to_felt252(to);
+            let amount_felt252: felt252 = amount.try_into().unwrap();
+            let hashed_value: felt252 = pedersen::pedersen(to_felt252, amount_felt252);
```

Comment:
contractAddress fits into felt252, but not amount. Here you're performing a hash, not on the original data, but on a truncating value. why not hashing the real amount (on 256 bits)?

Reply:
Yeah true, I had casted amount to felt252 because, the pedersen::pedersend requires (felt252,felt252) as inputs.

Reply:
I think we should hash the full amount 🤔 I will look into alexandria's merkle tree implementation but I think it can be improved 🤔

Reply:
imo you should 
```
let mut hash_state = pedersen(to_felt252, amount.low.into())
hash_state = pedersen(hash_state, amount.high.into())
```

that way you are hashing the correct data

Reply:
Ohh fine, I'll do this change